internal/pkg/scaffold/crd.go

```diff
@@ -132,8 +131,10 @@ func (s *CRD) CustomRender() ([]byte, error) {
 				if err = yaml.Unmarshal(cb, dstCRD); err != nil {
 					return nil, err
 				}
-				dstCRD.Spec.Validation = val
+				mergeCRDs(dstCRD, srcCRD)
```

Comment:
To prevent bugs like this happening in the future, I suggest we directly use the generated CRD's instead of trying to use old CRD data. Thoughts?

Reply:
Would there be anything in the old CRD data we would miss by using the generated CRDs? If not, that sounds good to me.

Reply:
Custom validation is the main issue. I submitted a [ticket upstream](https://github.com/kubernetes-sigs/controller-tools/issues/156) to merge existing and new validations, and have #1155 to do that in the SDK. Part of the issue is there's no good way to determine what a user has manually added and what is old in all fields; parsing `+kubebuilder` tagged types only partially solves the problem.

Other than validation, there's not much a user can add that isn't already autogenerated.

Reply:
> I suggest we directly use the generated CRD's instead of trying to use old CRD data

@estroz Do you mean just overwrite the entire CRD and not just the validation+printerColumns fields?
What if the user needs modifies other fields like the scope of the CRD.
Or adds in custom short names for the CRD.
https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.14/#customresourcedefinitionnames-v1beta1-apiextensions-k8s-io

I think we can't autogenerate that.

Reply:
@hasbro17 I believe most, if not all, CRD fields can be set using code annotations:
* scope: https://github.com/kubernetes-sigs/controller-tools/blob/master/pkg/internal/codegen/parse/crd.go#L76-L80
* short names (comma separated list): https://github.com/kubernetes-sigs/controller-tools/blob/master/pkg/internal/codegen/parse/index.go#L132-L133

That being said, there's no documentation on CRD generation in the SDK to actually tell users this is possible. I'm working on that.

Reply:
@estroz I'm fine with just overwriting the entire CRD if we have the documentation to point out all the kubebuilder comment annotations for the CRD.
Your call on if you want to add the documentation to this PR or if you want to open an issue and do a follow up.

Reply:
I'll do the former here so we can fix all issues going forward in one PR.